lib/modules/platform/gerrit/types.ts

```diff
@@ -0,0 +1,163 @@
+    | 'Cherry Pick'
+    | 'Rebase If Necessary'
+    | 'Rebase Always';
+  mergeable?: boolean;
+  submittable?: boolean;
+  insertions: number;
+  deletions: number;
+  total_comment_count?: number;
+  unresolved_comment_count?: number;
+  _number: number;
+  owner: GerritAccountInfo;
+  actions?: { [key: string]: GerritActionInfo };
+  submit_records: any[]; // SubmitRecordInfo[]
+  requirements?: any[]; //List of the requirements
+  submit_requirements?: any[]; //List of the SubmitRequirementResultInfo
+  labels?: { [key: string]: GerritLabelInfo };
+  permitted_labels?: any[];
+  removable_reviewers?: any[];
+  reviewers?: { [key in GerritReviewersType]: GerritAccountInfo[] }; //key = ReviewerState
+  pending_reviewers?: any;
+  reviewer_updates?: any;
+  messages?: GerritChangeMessageInfo[];
+  current_revision?: string;
+  revisions?: { [key: string]: GerritRevisionInfo }; //All patch sets of this change as a map that maps the commit ID of the patch set to a RevisionInfo entity.
+  meta_rev_id?: string;
+  tracking_ids?: any;
+  _more_changes?: any;
+  problems: any[];
+  is_private?: boolean;
+  work_in_progress?: boolean;
+  has_review_started?: boolean;
+  revert_of?: string;
+  submission_id?: string;
+  cherry_pick_of_change?: string;
+  cherry_pick_of_patch_set?: string;
+  contains_git_conflicts?: string;
+}
+
+//type ReviewerState = "REVIEWER"|"CC"|"REMOVED";
+export interface GerritRevisionInfo {
+  kind: string; //The change kind. Valid values are REWORK, TRIVIAL_REBASE, MERGE_FIRST_PARENT_UPDATE, NO_CODE_CHANGE, and NO_CHANGE.
+  _number: string; //The patch set number, or edit if the patch set is an edit.
+  created: Date; //TODO: map
+  uploader: GerritAccountInfo;
+  ref: string; //The Git reference for the patch set.
+  actions?: { [key: string]: GerritActionInfo };
+  //... many more...still not necessary
+}
+
+export interface GerritChangeMessageInfo {
+  id: string;
+  author?: GerritAccountInfo;
+  real_author?: GerritAccountInfo;
+  date: Date; //TODO: convert
+  message: string;
+  accountsInMessage: GerritAccountInfo[];
+  tag?: string;
+  _revision_number?: number;
+}
+
+export interface GerritLabelInfo {
+  optional?: boolean;
+  description?: string;
+  approved?: GerritAccountInfo;
+  rejected?: GerritAccountInfo;
+  recommended?: GerritAccountInfo;
+  disliked?: GerritAccountInfo;
+  blocking?: boolean;
+  value?: string; //The voting value of the user who recommended/disliked this label on the change if it is not “+1”/“-1”.
+  default_value?: string;
+  votes?: number[]; //A list of integers containing the vote values applied to this label at the latest patchset.
+  all?: any[];
+  values?: any; //A map of all values that are allowed for this label. The map maps the values (“-2”, “-1”, " `0`", “+1”, “+2”) to the value descriptions.
+}
+
+export interface GerritActionInfo {
+  method?: string;
+  label?: string;
+  title?: string;
+  enabled?: boolean;
+}
+
+export interface GerritAccountInfo {
```

Comment:
you should remove unrelated / unused fields from all interfaces to reduce possible confusion for later maintenance/ improvement

Reply:
sorry, but i don't really like this idea. 
I think they will not confuse but allow other devs to improve the current implementation based on other fields.

Reply:
> We're in progress to validate json responses with `zod`, so those unused properties will be removed anyways then.

What do you mean? 
Someone will create manually a zod-schema for all gerrit-REST responses to validate each response? 
Why?

Reply:
okay, i've stripped it down to the a minimum.